build.js

```diff
@@ -220,27 +364,21 @@ function patchLibs(patchTool, patchPath, target) {
   }
 }
 
-/**
- * Modify the provided environment object to be appropriate for cross-compilation for a darwin
- * target from a linux host
- */
-function patchEnvironment(env) {
-  if (!env.hasOwnProperty("CAPNP_LIBDIR")) {
-    console.error("CAPNP_LIBDIR must be set to a place I can find compiled libcapnp, libkj and friends for the target plat/arch");
-    process.exit(1);
-  }
+/*
+If we're using a host->target configuration other than linux->win32, run the above functions to
+build capnproto and `capnp.node`, then move `capnp.node` to the expected location.
 
-  if (!env.hasOwnProperty("CAPNP_INCDIR")) {
-    console.error("CAPNP_INCDIR must be set to a place I can find headers for libcapnp and libkj!");
-    process.exit(1);
-  }
+In the linux->win32 case, we're stuck using a precompiled binary, because (speaking from hard
+experience) `node-gyp` doesn't support that type of cross-compilation. We simply copy it from our
+`prebuilt/win32-x64` subdirectory.
+*/
 
-  return {
-    CC: "o64-clang",
-    CXX: "o64-clang++",
-    CXXFLAGS: "-mmacosx-version-min=10.7 -std=c++14 -stdlib=libc++ -I" + env.CAPNP_INCDIR,
-    LDFLAGS: "-L" + env.CAPNP_LIBDIR,
-    PATCH_TOOL: "x86_64-apple-darwin20.2-install_name_tool",
-    ...env
-  };
+if (process.platform === "linux" && args.targetPlatform === "win32") {
+  moveBuildResult('prebuilt/win32-x64/capnp.node', buildEnvironment);
```

Comment:
Logging that we're reusing a precompiled binary might be a helpful pointer if someone is debugging things in future.